File: php/backend/App_service/app/Http/Controllers/AuthController.php
<?php

namespace App\Http\Controllers;

use App\DTOs\LoginDTO;
use App\DTOs\SignupDTO;
use App\Http\Requests\LoginRequest;
use App\Http\Requests\SignupRequest;
use App\Services\AuthService;
use Illuminate\Http\JsonResponse;

class AuthController extends Controller
{
    public function __construct(private AuthService $auth) {}

    public function signup(SignupRequest $request): JsonResponse
    {
        $result = $this->auth->signup(SignupDTO::fromRequest($request->validated()));

        return $result['success']
            ? response()->json($result['user'], 201)
            : response()->json(['error' => $result['message']], 400);
    }

    public function login(LoginRequest $request): JsonResponse
    {
        $result = $this->auth->login(LoginDTO::fromRequest($request->validated()));

        return $result['success']
            ? response()->json(['token' => $result['token'], 'user' => $result['user']])
            : response()->json(['error' => $result['message']], 401);
    }
}

File: csharp/backend/CONTROLLER/ Authcontroller.cs
using LeadApi.Models;
using LeadApi.Services;

namespace LeadApi.Controllers;

public static class AuthController
{
    public static void MapAuthEndpoints(this WebApplication app)
    {
        var group = app.MapGroup("/api/auth");

        // group.MapPost("/signup", (SignupRequest request, AuthService auth) =>
        // {
        //     var user = auth.Signup(request);
        //     return Results.Created("/api/users", user);
        // });

        // group.MapPost("/login", (LoginRequest request, AuthService auth) =>
        // {
        //     var result = auth.Login(request);
        //     return result is { } r
        //         ? Results.Ok(new { token = r.Token, user = r.User })
        //         : Results.Unauthorized();
        // });

        group.MapPost("/signup", async (SignupRequest request, AuthService auth) =>
        {
            var (success, message, user) = await auth.SignupAsync(request);
            return success
                ? Results.Created("/api/users", user)
                : Results.BadRequest(new { message });
        });  
           // Results.Created("/api/users", await auth.SignupAsync(request))
        

        group.MapPost("/login", async (LoginRequest request, AuthService auth) =>
        {
            var result = await auth.LoginAsync(request);
            return result is { } r
                ? Results.Ok(new { token = r.Token, user = r.User })
                : Results.Unauthorized();
        });
    }
}